gef.py

```diff
@@ -2172,7 +2172,12 @@ def read_cstring_from_memory(address, max_length=GEF_MAX_STRING_LENGTH, encoding
 
     char_t = cached_lookup_type("char")
     char_ptr = char_t.pointer()
-    res = gdb.Value(address).cast(char_ptr).string(encoding=encoding).strip()
+    try:
+        res = gdb.Value(address).cast(char_ptr).string(encoding=encoding).strip()
+    except gdb.error:
+        length = min(address|(DEFAULT_PAGE_SIZE-1), max_length+1)
+        mem = bytes(read_memory(address, length)).decode('utf-8')
+        res = mem.split('\x00', 1)[0]
     res2 = res.replace('\n','\\n').replace('\r','\\r').replace('\t','\\t')
```

Comment:
We could use `.encode("unicode_escape")` here, though it might not work on python2

Reply:
No I mean `.encode("unicode_escape")` can get rid of the replace chain. I know it's not your line edit but let's fix it.

Reply:
Oh! Got it.